### PR TITLE
 Added support for `group_add` key

### PR DESCRIPTION
**Pull request by surajnarwade**

### CODE REVIEW

File: pkg/loader/compose/v1v2.go [hunk 1]
```diff
@@ -264,6 +264,16 @@ func libComposeToKomposeMapping(composeObject *project.Project) (kobject.Kompose
 		serviceConfig.MemLimit = composeServiceConfig.MemLimit
 		serviceConfig.TmpFs = composeServiceConfig.Tmpfs
 		serviceConfig.StopGracePeriod = composeServiceConfig.StopGracePeriod
+		var t2 []int64
+		for _, i := range composeServiceConfig.GroupAdd {
+			j, err := strconv.Atoi(i)
+			if err != nil {
+				panic(err)
```
Comment (cdrage) [inline pkg/loader/compose/v1v2.go]: needs to use errors.new for this

File: pkg/transformer/kubernetes/kubernetes.go [hunk 2]
```diff
@@ -103,8 +103,11 @@ func (k *Kubernetes) CheckUnsupportedKey(komposeObject *kobject.KomposeObject, u
 }
 
 // InitPodSpec creates the pod specification
-func (k *Kubernetes) InitPodSpec(name string, image string) api.PodSpec {
+func (k *Kubernetes) InitPodSpec(name string, image string, groupadd []int64) api.PodSpec {
```
Comment (cdrage) [inline pkg/transformer/kubernetes/kubernetes.go]: this is bad, no need to add groupadd here, we can add this to k8sutils.go, having it in kubernetes.go / openshift.go will make initpodspec messy.

File: script/test/fixtures/buildargs/output-os-template.json [hunk 3]
```diff
@@ -113,7 +113,8 @@
                 "resources": {}
               }
             ],
-            "restartPolicy": "Always"
+            "restartPolicy": "Always",
+	    "securityContext": {}
```
Comment (cdrage) [inline script/test/fixtures/buildargs/output-os-template.json]: all of the tests add `securityContext`, we should have an if statement in the code that ONLY adds securityContext if it's been supplied. no need for blank resources like this.

**Comment by surajnarwade** [comment]
I have updated PR with suggested changes, @cdrage @surajssd needs review on it

File: pkg/loader/compose/v1v2.go [hunk 4]
```diff
@@ -264,6 +264,16 @@ func libComposeToKomposeMapping(composeObject *project.Project) (kobject.Kompose
 		serviceConfig.MemLimit = composeServiceConfig.MemLimit
 		serviceConfig.TmpFs = composeServiceConfig.Tmpfs
 		serviceConfig.StopGracePeriod = composeServiceConfig.StopGracePeriod
+		var t2 []int64
+		for _, i := range composeServiceConfig.GroupAdd {
+			j, err := strconv.Atoi(i)
+			if err != nil {
+				return kobject.KomposeObject{}, errors.Wrap(err, "Unable to get group_add key")
```
Comment (containscafeine) [inline pkg/loader/compose/v1v2.go]: nit: start the error with lowercase

File: pkg/loader/compose/v1v2.go [hunk 5]
```diff
@@ -264,6 +264,16 @@ func libComposeToKomposeMapping(composeObject *project.Project) (kobject.Kompose
 		serviceConfig.MemLimit = composeServiceConfig.MemLimit
 		serviceConfig.TmpFs = composeServiceConfig.Tmpfs
 		serviceConfig.StopGracePeriod = composeServiceConfig.StopGracePeriod
+		var t2 []int64
```
Comment (containscafeine) [inline pkg/loader/compose/v1v2.go]: nit: what does t2 mean? How about a more explanatory variable name?

File: script/test/cmd/tests.sh [hunk 6]
```diff
@@ -174,6 +174,11 @@ convert::expect_success "kompose -f $KOMPOSE_ROOT/script/test/fixtures/tty-true/
 # openshift test
 convert::expect_success "kompose --provider openshift -f $KOMPOSE_ROOT/script/test/fixtures/tty-true/docker-compose.yml convert --stdout -j" "$KOMPOSE_ROOT/script/test/fixtures/tty-true/output-oc.json"
 
+# Test related to "group_add" in docker-compose
+# kubernetes test
+convert::expect_success "kompose -f $KOMPOSE_ROOT/script/test/fixtures/group-add/docker-compose.yml convert --stdout -j" "$KOMPOSE_ROOT/script/test/fixtures/group-add/output-k8s.json"
+# openshift test
+convert::expect_success "kompose --provider openshift -f $KOMPOSE_ROOT/script/test/fixtures/group-add/docker-compose.yml convert --stdout -j" "$KOMPOSE_ROOT/script/test/fixtures/group-add/output-os.json"
```
Comment (containscafeine) [inline script/test/cmd/tests.sh]: Can we add a failing test for both the providers? Makes the tests more robust.

File: script/test/fixtures/gitlab/docker-compose.yml [hunk 7]
```diff
@@ -4,7 +4,7 @@ services:
   postgresql:
     image: swordphilic/postgresql
     ports:
-      - "$DB_PORT"
+      - $DB_PORT
```
Comment (cdrage) [inline script/test/fixtures/gitlab/docker-compose.yml]: Why is this changed? Please revert back to with quotes.

File: pkg/loader/compose/v1v2.go [hunk 8]
```diff
@@ -264,6 +264,16 @@ func libComposeToKomposeMapping(composeObject *project.Project) (kobject.Kompose
 		serviceConfig.MemLimit = composeServiceConfig.MemLimit
 		serviceConfig.TmpFs = composeServiceConfig.Tmpfs
 		serviceConfig.StopGracePeriod = composeServiceConfig.StopGracePeriod
+		var groupAdd []int64
```
Comment (cdrage) [inline pkg/loader/compose/v1v2.go]: Please add spacing, newline here as well as a newline below (after `serviceConfig.GroupAdd = groupAdd`).

Since this is a range, perhaps it may be better to make this into a separate function? 

Also add a small comment please :)

**Comment by cdrage** [comment]
Please add a description to your commit as well as update this original PR description.

**Comment by surajnarwade** [comment]
@cdrage updated with suggested changes

**Comment by cdrage** [comment]
Awesome! This is great. Changes are good. Thanks @surajnarwade for the hard work.
